### DISCUSSION TITLE
Random Colors setting

### DISCUSSION THREAD

**Discussion by 3isenHeiM** [kimai/kimai#3446] [Questions & Answers]
Hi,
I wonder where do the "Random colors for unassigned objects" has an effect.

If I create a new activity (without color assigned) to a project (with red assigned), what's the expected behavior ?
Should it be random or Red ?

Currently every new object is gray...

**Comment by kevinpapst** [comment]
https://www.kimai.org/documentation/configurations.html#theme

I think only used in some places like the reports, not entirely sure where exactly. 

I have to decide whether to keep or remove this setting. Probably it is a good idea to remove it for #2902 .

**Reply by 3isenHeiM** [reply]
Well, manually choosing colors for customers, projects, activities is kinda overrated, but the option to automatically assign a random color is pretty neat !

You get used to the colors assigned...


On my instance, if I create a new customer, and leave the color un-determined, it stays in gray...

Here is my creating a new customer 
![Screenshot at 2022-07-29 14-41-00](https://user-images.githubusercontent.com/26417172/181763087-04d988cf-735b-400b-adfc-11da08344416.png)

Then in the list it stays in gray
![Screenshot at 2022-07-29 14-41-12](https://user-images.githubusercontent.com/26417172/181763116-9fe62c12-bcf2-4bb9-b015-72afb81c67bf.png)


And the setting is well enabled.

Have I understood that feature wrongly ? IMO the new customer should be in a random color...

**Reply by kevinpapst** [reply]
>  Well, manually choosing colors for customers, projects, activities is kinda overrated

For you ^^

> Have I understood that feature wrongly ?

1. You ignored what I wrote and what is written in the documentation. It is used in "some places" does not mean "everywhere".
2. This feature should not replace manual color choices. Instead it tries to fix a certain problem: if you have report e.g. "project detail" where e.g. a cake-chart shows users or activities and those do only have grey colors, you cannot identify the different graph sections. That's was the main use-case why the setting was added.  The graph sections get random colors. Check the tabs in the project detail reports. I think that's where it was used first.

**Reply by kevinpapst** [reply]
Thanks for your input. I removed the setting for the next major release #2902 and added the random color to all listings.
